L1Trigger/CSCTriggerPrimitives/src/CSCUpgradeMotherboardLUT.cc

```diff
@@ -0,0 +1,320 @@
+  , lut_wg_eta_even(0)
+  , lut_pt_vs_dphi_gemcsc(0)
+
+  , gem_roll_eta_limits_odd_l1(0)
+  , gem_roll_eta_limits_odd_l2(0)
+  , gem_roll_eta_limits_even_l1(0)
+  , gem_roll_eta_limits_even_l2(0)
+    
+  , csc_wg_to_gem_roll_odd_l1(0)
+  , csc_wg_to_gem_roll_odd_l2(0)
+  , csc_wg_to_gem_roll_even_l1(0) 
+  , csc_wg_to_gem_roll_even_l2(0)
+{
+}
+
+std::vector<std::pair<int,int> >*
+CSCGEMMotherboardLUT::get_csc_wg_to_gem_roll(Parity par, int layer) const
+{
+  if (par==Parity::Even){ return layer==1 ? csc_wg_to_gem_roll_even_l1 : csc_wg_to_gem_roll_even_l2; }
+  else                  { return layer==1 ? csc_wg_to_gem_roll_odd_l1 :  csc_wg_to_gem_roll_odd_l2; }
+} 
+
+std::vector<int>* 
+CSCGEMMotherboardLUTME11::get_gem_pad_to_csc_hs(Parity par, enum CSCPart p) const
+{
+  if (p==CSCPart::ME1A) { return par==Parity::Even ? gem_pad_to_csc_hs_me1a_even : gem_pad_to_csc_hs_me1a_odd; }
+  else                  { return par==Parity::Even ? gem_pad_to_csc_hs_me1b_even : gem_pad_to_csc_hs_me1b_odd; }
+}
+
+std::vector<int>* 
+CSCGEMMotherboardLUTME21::get_gem_pad_to_csc_hs(Parity par, enum CSCPart p) const
+{
+  return par==Parity::Even ? gem_pad_to_csc_hs_even : gem_pad_to_csc_hs_odd;
+}
+
+std::vector<std::pair<int,int> >* 
+CSCGEMMotherboardLUTME21::get_csc_hs_to_gem_pad(Parity par, enum CSCPart p) const
+{
+  return par==Parity::Even ? csc_hs_to_gem_pad_even : csc_hs_to_gem_pad_odd;
+}
+
+std::vector<std::pair<int,int> >* 
+CSCGEMMotherboardLUTME11::get_csc_hs_to_gem_pad(Parity par, enum CSCPart p) const
+{
+  if (p==CSCPart::ME1A) { return par==Parity::Even ? csc_hs_to_gem_pad_me1a_even : csc_hs_to_gem_pad_me1a_odd; }
+  else                  { return par==Parity::Even ? csc_hs_to_gem_pad_me1b_even : csc_hs_to_gem_pad_me1b_odd; }
+}
+
+std::vector<std::vector<double>> * 
+CSCGEMMotherboardLUTME11::get_lut_wg_vs_hs(enum CSCPart p) const
+{
+  if (p==CSCPart::ME1A)      { return lut_wg_vs_hs_me1a;  }
+  else if (p==CSCPart::ME1B) { return lut_wg_vs_hs_me1b;  }
+  else                       { return lut_wg_vs_hs_me1ag; } 
+}
+
+CSCGEMMotherboardLUT::~CSCGEMMotherboardLUT(){
+  if(lut_wg_eta_odd       ) delete lut_wg_eta_odd       ;
+  if(lut_wg_eta_even      ) delete lut_wg_eta_even      ;
+  if(lut_pt_vs_dphi_gemcsc) delete lut_pt_vs_dphi_gemcsc;
+
+  if(gem_roll_eta_limits_odd_l1) delete gem_roll_eta_limits_odd_l1;
+  if(gem_roll_eta_limits_odd_l2) delete gem_roll_eta_limits_odd_l2;
+  if(gem_roll_eta_limits_even_l1) delete gem_roll_eta_limits_even_l1;
+  if(gem_roll_eta_limits_even_l2) delete gem_roll_eta_limits_even_l2;
+
+ if (csc_wg_to_gem_roll_odd_l1) delete csc_wg_to_gem_roll_odd_l1;
+ if (csc_wg_to_gem_roll_odd_l2) delete csc_wg_to_gem_roll_odd_l2;
+ if (csc_wg_to_gem_roll_even_l1) delete csc_wg_to_gem_roll_even_l1; 
+ if (csc_wg_to_gem_roll_even_l2) delete csc_wg_to_gem_roll_even_l2;
+}
+
+
+CSCGEMMotherboardLUTME11::CSCGEMMotherboardLUTME11() 
+  : CSCGEMMotherboardLUT()
+{
+  lut_wg_eta_odd = new std::vector<std::vector<double>> {
```

Comment:
If there's no reason to allocate this dynamically, then just using a regular vector is preferable